components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -435,6 +456,50 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_replace() {
+        let cases = vec![
+            (vec![None, Some("a"), Some("b")], None),
+            (vec![Some("a"), None, Some("b")], None),
+            (vec![Some("a"), Some("b"), None], None),
+            (
+                vec![Some("www.mysql.com"), Some("mysql"), Some("pingcap")],
+                Some("www.pingcap.com"),
+            ),
+            (
+                vec![Some("www.mysql.com"), Some("w"), Some("1")],
+                Some("111.mysql.com"),
+            ),
+            (vec![Some("1234"), Some("2"), Some("55")], Some("15534")),
+            (vec![Some(""), Some("a"), Some("b")], Some("")),
+            (vec![Some("abc"), Some(""), Some("d")], Some("abc")),
+            (vec![Some("aaa"), Some("a"), Some("")], Some("")),
+            (vec![Some("aaa"), Some("a"), Some("")], Some("")),
+            (
+                vec![Some("新年快乐"), Some("年"), Some("春")],
+                Some("新春快乐"),
+            ),
+            (
+                vec![Some("心想事成"), Some("心"), Some("❤️")],
+                Some("❤️想事成"),
+            ),
+        ];
+
+        for (args, expect_output) in cases {
+            let params = args.iter().map(|arg| arg.map(|s| s.as_bytes().to_vec()));
+            let output = RpnFnScalarEvaluator::new()
+                .push_params(params)
+                .evaluate(ScalarFuncSig::Replace)
+                .unwrap();
```

Comment:
Since `replace` has constant argument length. I prefer to have `args` be a tuple of arguments, which can, therefore, be mapped and pushed individually.